### ISSUE TITLE
`copy_to_bytes` consumes &self

### ISSUE THREAD

**Issue by o0Ignition0o** [tokio-rs/bytes#473]
Hey everyone,

first of all I'd like to thank you for the tremendous work it took to reach 1.0 🎉 

I was quite confused during a webapp's tokio upgrade path because one of my proxies would return empty (warp) response bodies after the migration.

After longer than I'd like to admit ( 😅 ) I zeroed in on the fact that `copy_to_bytes()` took `&mut self`, and had a look at [the documentation](https://docs.rs/bytes/1.0.1/bytes/buf/trait.Buf.html#method.copy_to_bytes):

`Consumes len bytes inside self and returns new instance of Bytes with this data.`

It confused me quite a bit because that's not the behavior I would expect from `copy_to*` methods.

There's probably a rationale behind that, and if you might have already considered `move_to` `transfer_to` and other variants, and decided otherwise, feel free to close the issue.

The fact it might confuse others as well prompted me to open this issue, and I'd gladly help rename the functions if it makes sense!

Thank's again for your time :)

**Comment by Darksonn** [comment]
They can't just be renamed due to backwards compatibility. The most we could do is deprecate and `#[doc(hidden)]` it, adding an alternative.

**Comment by o0Ignition0o** [comment]
Oh I see, thanks a lot for your reply!

**Comment by carllerche** [comment]
That is pretty unfortunate... we will probably need to add an alternative :'(

**Comment by carllerche** [comment]
Actually, wait... it takes `&mut self` because it advances the internal cursor.

**Comment by carllerche** [comment]
Given that `&mut self` is *correct* and it is mostly a less-than-ideal name, I don't think we should deprecate & create an alternative. We should just document the method and deal with it.

**Comment by seanmonstar** [comment]
Yea, the question of the name for this method came up when it was renamed, and back then we noticed the same: the name isn't great, but there weren't any other better names.

**Comment by o0Ignition0o** [comment]
> I don't think we should deprecate & create an alternative. We should just document the method and deal with it.

Yeah that makes sense, it is documented already so I guess there's not much more we can do then ^^'

**Comment by nazar-pc** [comment]
`read_to_bytes()` would hint that there is an internal cursor that is being advanced.
